### ISSUE TITLE
Bug: `prefer-regex-literals` rule matches more scenarios

### ISSUE THREAD

**Issue by btea** [eslint/eslint#19482]
### Environment

Node version: 22.5.1
npm version: 10.8.2
Local ESLint version: 9.21.0
Global ESLint version: 9.21.0
Operating System: win10

### What parser are you using?

Default (Espree)

### What did you do?

I hope the `prefer-regex-literals` rule can match more cases. Refer to the link below.
<details>
<summary>Configuration</summary>

```js
<!-- Paste your configuration here -->
```
</details>

```js
<!-- Paste your code here -->
```

### What did you expect to happen?

```js
const a = new RegExp('a') // This is invalid
const b = new RegExp('a' + 'b') // This is invalid
```

### What actually happened?

```js
const a = new RegExp('a') // This is invalid
const b = new RegExp('a' + 'b') // This is valid
```

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgYSA9IG5ldyBSZWdFeHAoJ2EnKVxuY29uc3QgYiA9IG5ldyBSZWdFeHAoJ2EnICsgJ2InKSAvLyBJIHRoaW5rIHRoaXMgcmVndWxhciBleHByZXNzaW9uIHNob3VsZCBhbHNvIGdpdmUgdGhlIHNhbWUgZXJyb3IgbWVzc2FnZSBhcyBhYm92ZS4iLCJvcHRpb25zIjp7InJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjpbImVycm9yIl0sImZvci1kaXJlY3Rpb24iOlsiZXJyb3IiXSwiZ2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjpbImVycm9yIl0sIm5vLWNhc2UtZGVjbGFyYXRpb25zIjpbImVycm9yIl0sIm5vLWNsYXNzLWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb21wYXJlLW5lZy16ZXJvIjpbImVycm9yIl0sIm5vLWNvbmQtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb25zdGFudC1iaW5hcnktZXhwcmVzc2lvbiI6WyJlcnJvciJdLCJuby1jb25zdGFudC1jb25kaXRpb24iOlsiZXJyb3IiXSwibm8tY29udHJvbC1yZWdleCI6WyJlcnJvciJdLCJuby1kZWJ1Z2dlciI6WyJlcnJvciJdLCJuby1kZWxldGUtdmFyIjpbImVycm9yIl0sIm5vLWR1cGUtYXJncyI6WyJlcnJvciJdLCJuby1kdXBlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXSwibm8tZHVwZS1lbHNlLWlmIjpbImVycm9yIl0sIm5vLWR1cGUta2V5cyI6WyJlcnJvciJdLCJuby1kdXBsaWNhdGUtY2FzZSI6WyJlcnJvciJdLCJuby1lbXB0eSI6WyJlcnJvciJdLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tZW1wdHktcGF0dGVybiI6WyJlcnJvciJdLCJuby1lbXB0eS1zdGF0aWMtYmxvY2siOlsiZXJyb3IiXSwibm8tZXgtYXNzaWduIjpbImVycm9yIl0sIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6WyJlcnJvciJdLCJuby1mYWxsdGhyb3VnaCI6WyJlcnJvciJdLCJuby1mdW5jLWFzc2lnbiI6WyJlcnJvciJdLCJuby1nbG9iYWwtYXNzaWduIjpbImVycm9yIl0sIm5vLWltcG9ydC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW52YWxpZC1yZWdleHAiOlsiZXJyb3IiXSwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOlsiZXJyb3IiXSwibm8tbG9zcy1vZi1wcmVjaXNpb24iOlsiZXJyb3IiXSwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tbmV3LW5hdGl2ZS1ub25jb25zdHJ1Y3RvciI6WyJlcnJvciJdLCJuby1ub25vY3RhbC1kZWNpbWFsLWVzY2FwZSI6WyJlcnJvciJdLCJuby1vYmotY2FsbHMiOlsiZXJyb3IiXSwibm8tb2N0YWwiOlsiZXJyb3IiXSwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjpbImVycm9yIl0sIm5vLXJlZGVjbGFyZSI6WyJlcnJvciJdLCJuby1yZWdleC1zcGFjZXMiOlsiZXJyb3IiXSwibm8tc2VsZi1hc3NpZ24iOlsiZXJyb3IiXSwibm8tc2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6WyJlcnJvciJdLCJuby1zcGFyc2UtYXJyYXlzIjpbImVycm9yIl0sIm5vLXRoaXMtYmVmb3JlLXN1cGVyIjpbImVycm9yIl0sIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjpbImVycm9yIl0sIm5vLXVucmVhY2hhYmxlIjpbImVycm9yIl0sIm5vLXVuc2FmZS1maW5hbGx5IjpbImVycm9yIl0sIm5vLXVuc2FmZS1uZWdhdGlvbiI6WyJlcnJvciJdLCJuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmciOlsiZXJyb3IiXSwibm8tdW51c2VkLWxhYmVscyI6WyJlcnJvciJdLCJuby11bnVzZWQtcHJpdmF0ZS1jbGFzcy1tZW1iZXJzIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtYmFja3JlZmVyZW5jZSI6WyJlcnJvciJdLCJuby11c2VsZXNzLWNhdGNoIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtZXNjYXBlIjpbImVycm9yIl0sIm5vLXdpdGgiOlsiZXJyb3IiXSwicmVxdWlyZS15aWVsZCI6WyJlcnJvciJdLCJ1c2UtaXNuYW4iOlsiZXJyb3IiXSwidmFsaWQtdHlwZW9mIjpbImVycm9yIl0sInByZWZlci1yZWdleC1saXRlcmFscyI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### Participation

- [x] I am willing to submit a pull request for this issue.

**Comment by ntnyq** [comment]
This is an intended behavior. As:

https://github.com/eslint/eslint/blob/fbdeff08f3bf4edd7f686af91d9ac0ed6e295080/tests/lib/rules/prefer-regex-literals.js#L51

**Comment by btea** [comment]
Yeah, I noticed this, but I don't know why. I think `new RegExp('a' + 'b')` should be considered equal to `new RegExp('ab')`.

**Comment by mdjermanovic** [comment]
This is intended behavior. Per the [documentation for this rule](https://eslint.org/docs/latest/rules/prefer-regex-literals#rule-details):

> This rule disallows the use of the `RegExp` constructor function with string literals as its arguments.
> 
> This rule also disallows the use of the `RegExp` constructor function with template literals without expressions
> and `String.raw` tagged template literals without expressions.
> 
> The rule does not disallow all use of the `RegExp` constructor. It should be still used for
> dynamically generated regular expressions.

In the `new RegExp('a' + 'b')` example, the argument is an expression that doesn't fall under the above.

That said, I wouldn't be opposed to adding an option for this pattern. @eslint/eslint-team thoughts?

**Comment by mdjermanovic** [comment]
Ah, yes, combination with `no-useless-concat` would address this. I agree with @nzakas. Closing as this is not a bug but the intended behavior of this rule, and we're not in favor of adding a new option.

**Comment by btea** [comment]
Thanks for your reply, it seems I am not familiar with these rules.
